### ISSUE TITLE
Чем клиент отличается от сессии?

### ISSUE THREAD

**Issue by bd100b** [AnisovAleksey/HamsterKombatBot#12]
Подскажите пожалуйста чем клиент отличается от сессии?

**Comment by AnisovAleksey** [comment]
Сессия - это авторизация в телеграме (и через нее дальнейшая авторизация в хомяке)
Клиент - токен авторизации для хомяка напрямую. Достать его бывает сложнее, но таким образом у тебя токен сгенерен будет через официальные пути, а не через левый апликейшен в телеграме, как это делалось с сессией

По сути с сессией ты имеешь доступ к аккаунту в тг, что как бы слишком много для этого бота)

**Comment by bd100b** [comment]
> Сессия - это авторизация в телеграме (и через нее дальнейшая авторизация в хомяке) Клиент - токен авторизации для хомяка напрямую. Достать его бывает сложнее, но таким образом у тебя токен сгенерен будет через официальные пути, а не через левый апликейшен в телеграме, как это делалось с сессией
> 
> По сути с сессией ты имеешь доступ к аккаунту в тг, что как бы слишком много для этого бота)

понял, спасибо. А еще вижу что вы в фингерпринтах только генерируете уникальный visitor_id а все остальные параметры по умолчанию. Это потому что все остальные не сильно влияют на уникальность или все же лучше и их рандомизировать?

**Comment by Yorksik** [comment]
> Сессия - это авторизация в телеграме (и через нее дальнейшая авторизация в хомяке) Клиент - токен авторизации для хомяка напрямую. Достать его бывает сложнее, но таким образом у тебя токен сгенерен будет через официальные пути, а не через левый апликейшен в телеграме, как это делалось с сессией
> 
> По сути с сессией ты имеешь доступ к аккаунту в тг, что как бы слишком много для этого бота)

Есть ли возможность достать его из под андроид без рута?

**Comment by AnisovAleksey** [comment]
> > Сессия - это авторизация в телеграме (и через нее дальнейшая авторизация в хомяке) Клиент - токен авторизации для хомяка напрямую. Достать его бывает сложнее, но таким образом у тебя токен сгенерен будет через официальные пути, а не через левый апликейшен в телеграме, как это делалось с сессией
> > По сути с сессией ты имеешь доступ к аккаунту в тг, что как бы слишком много для этого бота)
> 
> понял, спасибо. А еще вижу что вы в фингерпринтах только генерируете уникальный visitor_id а все остальные параметры по умолчанию. Это потому что все остальные не сильно влияют на уникальность или все же лучше и их рандомизировать?

С токеном этот фингерпринт не используется. visitor_id должен генерироваться на основе фингерпринта, но логику из клиента я выдрать не смог, поэтому лучше просто не использовать авторизацию через сессии

**Comment by AnisovAleksey** [comment]
> > Сессия - это авторизация в телеграме (и через нее дальнейшая авторизация в хомяке) Клиент - токен авторизации для хомяка напрямую. Достать его бывает сложнее, но таким образом у тебя токен сгенерен будет через официальные пути, а не через левый апликейшен в телеграме, как это делалось с сессией
> > По сути с сессией ты имеешь доступ к аккаунту в тг, что как бы слишком много для этого бота)
> 
> Есть ли возможность достать его из под андроид без рута?

Про андроид не подскажу, попробуйте поискать в сторону WebView inspecting

**Comment by bd100b** [comment]
> С токеном этот фингерпринт не используется. visitor_id должен генерироваться на основе фингерпринта, но логику из клиента я выдрать не смог, поэтому лучше просто не использовать авторизацию через сессии

правильно ли я понимаю что в таком случае фингерпринт особой роли не играет? если не использовать авторизацию через сессии

**Comment by AnisovAleksey** [comment]
> правильно ли я понимаю что в таком случае фингерпринт особой роли не играет? если не использовать авторизацию через сессии

Да, фингерпринт нужен для того чтобы получить токен с помощью телеграм сессии

**Comment by XPacL** [comment]
> > Сессия - это авторизация в телеграме (и через нее дальнейшая авторизация в хомяке) Клиент - токен авторизации для хомяка напрямую. Достать его бывает сложнее, но таким образом у тебя токен сгенерен будет через официальные пути, а не через левый апликейшен в телеграме, как это делалось с сессией
> > По сути с сессией ты имеешь доступ к аккаунту в тг, что как бы слишком много для этого бота)
> 
> Есть ли возможность достать его из под андроид без рута?

Единственный способ, что я накопал на просторах - это вытащить tgnet.dat из корневой директории телеги. Проблема в том что нужен рут, а рутировать нельзя по определенным причинам.

**Comment by AnisovAleksey** [comment]
> 
> Единственный способ, что я накопал на просторах - это вытащить tgnet.dat из корневой директории телеги. Проблема в том что нужен рут, а рутировать нельзя по определенным причинам.

Он вроде про токен спрашивал, а ты пишешь про сессию тг)

**Comment by XPacL** [comment]
Тогда я вообще не понимаю что это за токен и как его выдернуть. В сети вообще никакой информации нет. Только токены бота через BotFather нахожу как сделать и всё(((.

**Comment by AnisovAleksey** [comment]
https://github.com/mudachyo/Hamster-Kombat/tree/main
вот тут есть способ запуска хомяка в браузере. Там уже через developer tools можно посмотреть запросы на сервер и в них будет токен для авторизации

**Comment by bd100b** [comment]
> Да, фингерпринт нужен для того чтобы получить токен с помощью телеграм сессии

еще хотел уточнить, есть ли смысл делать под каждый аккаунт уникальный юзер агент если все равно буду использовать прокси?

**Comment by XPacL** [comment]
> https://github.com/mudachyo/Hamster-Kombat/tree/main вот тут есть способ запуска хомяка в браузере. Там уже через developer tools можно посмотреть запросы на сервер и в них будет токен для авторизации

Спасибо! Через этот способ нашёл токен "Auth Bearer". Это он?

**Comment by AnisovAleksey** [comment]
> еще хотел уточнить, есть ли смысл делать под каждый аккаунт уникальный юзер агент если все равно буду использовать прокси?

думаю есть смысл, но правды мы не узнаем) у меня везде один юзер агент, не парюсь особо

**Comment by AnisovAleksey** [comment]
> > https://github.com/mudachyo/Hamster-Kombat/tree/main вот тут есть способ запуска хомяка в браузере. Там уже через developer tools можно посмотреть запросы на сервер и в них будет токен для авторизации
> 
> Спасибо! Через этот способ нашёл токен "Auth Bearer". Это он?

Да

**Comment by bd100b** [comment]
> > > https://github.com/mudachyo/Hamster-Kombat/tree/main вот тут есть способ запуска хомяка в браузере. Там уже через developer tools можно посмотреть запросы на сервер и в них будет токен для авторизации
> > 
> > 
> > Спасибо! Через этот способ нашёл токен "Auth Bearer". Это он?
> 
> Да

разве он будет действителен постоянно? Там же есть какой то тайм то лив после которого надо заново проходить авторизацию или я не прав?

**Comment by AnisovAleksey** [comment]
> > > > https://github.com/mudachyo/Hamster-Kombat/tree/main вот тут есть способ запуска хомяка в браузере. Там уже через developer tools можно посмотреть запросы на сервер и в них будет токен для авторизации
> > > 
> > > 
> > > Спасибо! Через этот способ нашёл токен "Auth Bearer". Это он?
> > 
> > 
> > Да
> 
> разве он будет действителен постоянно? Там же есть какой то тайм то лив после которого надо заново проходить авторизацию или я не прав?

Пока сессия в тг активна - токен работает. Экспайр тайма нет

**Comment by XPacL** [comment]
> Нашёл способ, ура. Ты натолкнул на webview и прям помогло

Не помешали бы подробности)

**Comment by AnisovAleksey** [comment]
в телеграмме можно включить web view inspecting зажав на номер версии в настройках два раза. Тогда можно будет с хрома подключиться к вебвью

**Comment by XPacL** [comment]
Огонь. Очень полезная инфа, спасибо!

**Comment by AnisovAleksey** [comment]
Про хедеры и юзерагент я уже писал - сейчас слепок с иос девайса. По хорошему надо реализовать рандомную генерацию UA, либо руками в конфиге указывать. Но например в андроиде больше хедеров указывается, поэтому при замене UA - надо еще список хедеров сопоставлять

**Comment by AnisovAleksey** [comment]
> > Про хедеры и юзерагент я уже писал - сейчас слепок с иос девайса. По хорошему надо реализовать рандомную генерацию UA, либо руками в конфиге указывать. Но например в андроиде больше хедеров указывается, поэтому при замене UA - надо еще список хедеров сопоставлять
> 
> Я ручками меняю каждый раз, добавляю все атрибуты, которых нет. Тут теперь интересно, можно ли как-то через chrome devtools вытащить fingerprint. Просто я максимально хотел бы шифроваться под свой девайс

Если ты используешь токен, то фингерпринт не используется

**Comment by AnisovAleksey** [comment]
Если оформишь в пул реквест - будет здорово)

**Comment by dntaya** [comment]
> > Если оформишь в пул реквест - будет здорово)
> 
> Сделаем.
> 
> @XPacL , если будешь воровать UA, не забудь IP поменять с дебажного на девайсный. Тут `.... (KHTML, like Gecko) Version/4.0 Chrome/125.0.6422.165 Mobile Safari/537.36`| **При дебаге:** 125.0.6422.165 **На самом девайсе**: 125.0.0.0

это версия хрома
